FetchRequests/Sources/OrderedSetCompat.swift

```diff
@@ -0,0 +1,282 @@
+
+import Collections
+
+typealias OrderedSet = Collections.OrderedSet
+
+#else
+
+import Foundation
+
+struct OrderedSet<Element: Hashable> {
+    private(set) var elements: [Element] {
+        didSet {
+            reindex()
+        }
+    }
+    private(set) var unordered: Set<Element>
+    private var indexed: [Element: Int]
+
+    init() {
+        elements = []
+        unordered = []
+        indexed = [:]
+    }
+
+    init<S: Sequence>(_ sequence: S) where S.Element == Element {
+        self.init()
+        sequence.forEach { insert($0) }
+    }
+}
+
+// MARK: - Capacity
+
+extension OrderedSet {
+    mutating func removeAll(keepingCapacity keepCapacity: Bool = false) {
+        elements.removeAll(keepingCapacity: keepCapacity)
+        unordered.removeAll(keepingCapacity: keepCapacity)
+    }
+
+    mutating func reserveCapacity(_ minimumCapacity: Int) {
+        elements.reserveCapacity(minimumCapacity)
+        unordered.reserveCapacity(minimumCapacity)
+    }
+
+    var capacity: Int {
+        elements.capacity
+    }
+}
+
+// MARK: - Equatable
+
+extension OrderedSet: Equatable {
+    static func == (lhs: Self, rhs: Self) -> Bool {
+        lhs.elements == rhs.elements
+    }
+}
+
+// MARK: - Hashable
+
+extension OrderedSet: Hashable {
+    func hash(into hasher: inout Hasher) {
+        elements.hash(into: &hasher)
+    }
+}
+
+// MARK: - CustomStringConvertible
+
+extension OrderedSet: CustomStringConvertible {
+    var description: String {
+        elements.description
+    }
+}
+
+// MARK: - ExpressibleByArrayLiteral
+
+extension OrderedSet: ExpressibleByArrayLiteral {
+    init(arrayLiteral elements: Element...) {
+        self.init(elements)
+    }
+}
+
+// MARK: - SetAlgebra
+
+extension OrderedSet: SetAlgebra {
+    func contains(_ member: Element) -> Bool {
+        unordered.contains(member)
+    }
+
+    @discardableResult
+    mutating func remove(_ member: Element) -> Element? {
+        guard contains(member), let index = index(of: member) else {
+            return nil
+        }
+        elements.remove(at: index)
+        return unordered.remove(member)
+    }
+
+    @discardableResult
+    mutating func update(with newMember: Element) -> Element? {
+        // Unconditionally update our set, even if our set contains it
+
+        if contains(newMember), let index = index(of: newMember) {
+            elements[index] = newMember
+            return unordered.update(with: newMember)
+        } else {
+            insert(newMember)
+            return nil
+        }
+    }
+
+    @discardableResult
+    mutating func insert(
+        _ newMember: Element
+    ) -> (inserted: Bool, memberAfterInsert: Element) {
+        // Conditionally update our set, iff our set does not contain it
+
+        guard contains(newMember) else {
+            elements.append(newMember)
+            unordered.insert(newMember)
+            return (true, newMember)
+        }
+
+        // returns (false, oldMember)
```

Comment:
Delete this

Reply:
This is intentional, but I'll tweak the comment